extensions/amp-audioburst-audio/0.1/amp-audioburst-audio.js

```diff
@@ -0,0 +1,460 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds an Audioburst audio player
+ * @author Audioburst
+ *
+ * Example:
+ * <code>
+ * <amp-audioburst-audio
+ *  autoplay
+ *  layout="fixed-height"
+ *  height="315"
+ *  src="https://sapi.audioburst.com/audio/repo/play/web/l1AXp9wBR57y.mp3"
+ *  fullShow="http://storageaudiobursts.blob.core.windows.net/temp/11760_2018081122_t.mp3"
+ *  fullShowPosition="640">
+ * </amp-audioburst-audio>
+ */
+
+import {Layout} from '../../../src/layout';
+import {Services} from '../../../src/services';
+import {listen} from '../../../src/event-helper';
+import {setStyle, setStyles} from '../../../src/style';
+
+
+export class AmpAudioburstAudio extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private */
+    this.viewport_ = null;
+    /** @private {string} */
+    this.burstSrc_ = undefined;
+    /** @private {string} */
+    this.fullShowSrc_ = undefined;
+    /** @private {number} */
+    this.fullShowPosition_ = 0;
+    /** @public {boolean} */
+    this.isPlaying = false;
+    /** @public {boolean} */
+    this.ended = false;
+    /** @public {number} */
+    this.currentTime = 0;
+    /** @private {number} */
+    this.burstSaveTime_ = undefined;
+    /** @public {number} */
+    this.duration = 0;
+
+    /** @private {!Element} */
+    this.container_ = this.win.document.createElement('div');
+
+    /** @private {!Element} */
+    this.audio_ = null;
+
+    /** @private {!Element} */
+    this.pauseBtn_ = null;
+
+    /** @private {!Element} */
+    this.playBtn_ = null;
+
+    /** @private {!Element} */
+    this.currentTimeEl_ = null;
+
+    /** @private {!Element} */
+    this.durationEl_ = null;
+
+    /** @private {!Element} */
+    this.slider_ = null;
+  }
+
+  /** @override */
+  /*buildCallback() {
+    // this.container_.textContent = this.myText_;
+    // this.element.appendChild(this.container_);
+    // this.applyFillContent(this.container_, true);
+  }*/
+
+  /** @override */
+  layoutCallback() {
```

Comment:
You should move the element creation code into `buildCallback` and keep anything that consumes resources like loads and listeners into `layoutCallback`

Reply:
Due to specific of component (it's audio player, see amp-audio), element creation and consuming resources is very hard bound together. If I'll try to split it, the will become completely unreadable. If it's possible, please, may I this structure?